src/tables/filter.rs

```
@@ -0,0 +1,250 @@
+        let key = Filter::create_key(&self.username, &self.name);
+        let value = Value {
+            directions: self.directions,
+            keywords: self.keywords,
+            network_tags: self.network_tags,
+            customers: self.customers,
+            endpoints: self.endpoints,
+            sensors: self.sensors,
+            os: self.os,
+            devices: self.devices,
+            host_names: self.host_names,
+            user_ids: self.user_ids,
+            user_names: self.user_names,
+            user_departments: self.user_departments,
+            countries: self.countries,
+            categories: self.categories,
+            levels: self.levels,
+            kinds: self.kinds,
+            learning_methods: self.learning_methods,
+            confidence: self.confidence,
+        };
+        let value = super::serialize(&value)?;
+        Ok((key, value))
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct Value {
+    directions: Option<Vec<FlowKind>>,
+    keywords: Option<Vec<String>>,
+    network_tags: Option<Vec<String>>,
+    customers: Option<Vec<String>>,
+    endpoints: Option<Vec<FilterEndpoint>>,
+    sensors: Option<Vec<String>>,
+    os: Option<Vec<String>>,
+    devices: Option<Vec<String>>,
+    host_names: Option<Vec<String>>,
+    user_ids: Option<Vec<String>>,
+    user_names: Option<Vec<String>>,
+    user_departments: Option<Vec<String>>,
+    countries: Option<Vec<String>>,
+    categories: Option<Vec<u8>>,
+    levels: Option<Vec<u8>>,
+    kinds: Option<Vec<String>>,
+    learning_methods: Option<Vec<LearningMethod>>,
+    confidence: Option<f32>,
+}
+
+impl FromKeyValue for Filter {
+    fn from_key_value(key: &[u8], value: &[u8]) -> Result<Self> {
+        use anyhow::anyhow;
+
+        let sep = key
+            .iter()
+            .position(|c| *c == 0)
+            .ok_or(anyhow!("corruptted access token"))?;
+        let username = String::from_utf8_lossy(&key[..sep]).into_owned();
```

Shouldn't we report an error if the binaries from the database are invalid UTF-8?